### ISSUE TITLE
Update image generation section in compose.py

### ISSUE THREAD

**Issue by NikosAlexandris** [NikosAlexandris/gee_sar_fetcher#3]
See quick test script https://github.com/NikosAlexandris/gee_sar_fetcher/blob/9244fb6f06e11f42683aef591168a9dc972b0d95/geesarfetcher/test_compose_function.py.

Fix section that generate the final (image) data structure in `compose.py`: https://github.com/NikosAlexandris/gee_sar_fetcher/blob/9244fb6f06e11f42683aef591168a9dc972b0d95/geesarfetcher/compose.py#L350-L392

**Comment by NikosAlexandris** [comment]
@dimartinot Can you please help out?

**Comment by dimartinot** [comment]
I'd be happy to. 
Which error are you encoutering ? To be able to import the `compose` function from your test file, I needed to move it to the root of the git directory, at the same level as the README file.

Additionally, once this was done, the `timestamps` variable appeared to be missing in your compose function.

For that matter, you would just need to add in [this line](https://github.com/NikosAlexandris/gee_sar_fetcher/blob/9244fb6f06e11f42683aef591168a9dc972b0d95/geesarfetcher/compose.py#L325) the following code:
```python
timestamps = np.unique(
        [
            datetime.fromtimestamp(pixel_values[i]['timestamps'][j]).date()
            for i in range(len(pixel_values))
            for j in range(len(pixel_values[i]['timestamps']))
        ]
)
```

**Comment by NikosAlexandris** [comment]
I am inside a virtual environment, with all requirements installed. For the matter, I prefer to install packages with `-e`, i.e.,
```
cd geesarfetcher
pip install -e .
```
Then, from the package's root directory, I run the test script `test_compose_function.py` also like:
```
ipython -i geesarfetcher/test_compose_function.py
```

The `compose()` function won't deal with timestamps. Because the final product is a single composite image, it will store in the metadata the `start_date` and `end_date` for reference.

The original https://github.com/dimartinot/gee_sar_fetcher/blob/a78c42adad9ab35b4d0e7e4ad181adbd9c92fffa/geesarfetcher/__init__.py#L33 function has to deal with the timestamps. I see that the `indexes` https://github.com/dimartinot/gee_sar_fetcher/blob/a78c42adad9ab35b4d0e7e4ad181adbd9c92fffa/geesarfetcher/__init__.py#L244 are then defined based on the timestamps.

Since `compose()` won't need this, I have to re-think how to fill the final data structure (like `img`, `stack`, etc. as per the original `fetch()` function.

Does this make sense?  I would appreciate any help to design this in a good way. The goal is to have an output data structure that can be further re-used easily to do more math, export as a CSV and create plots.

**Comment by dimartinot** [comment]
I understand your point fully ! 
I define the indexes based on the timestamps for the reason that for some coordinates, at the edges of the satellite trajectory, they may have 2 values for the same day, which goes in conflict with most of the other pixels. For that matter, I average the two values to keep only one per available acquisition date.
Hence, the code that you highlight [here](https://github.com/NikosAlexandris/gee_sar_fetcher/issues/3#issue-745598302) is used to alleviate this problem.

If your end goal is a composite image, you do not have to loop over timestamp and can then reduce it using temporal average like this:
```python
 for p in tqdm(pixel_values): 
     x, y = latitudes_dictionary[p["lat"]], longitudes_dictionary[p["lon"]] 
     image[x, y, 0, :] = np.nanmean( p["VV"] )
     image[x, y, 1, :] = np.nanmean( p["VH"] )
```

I hope this matches your objective. 
I'd be happy to discuss more about the design of the returned object. I opted for a simple Python dictionnary because they are highly performant Data Structures. If you have another idea in mind, I'd be happy to discuss it.

Additionally, I had experimentally implemented the support for saving the "_image_" of each timestamp as a numpy file (i.e _.npy_), it could be interesting to provide such tool and it may match your expectations ?

**Comment by NikosAlexandris** [comment]
I got it working.

For a CSV output, it might be easier to build an CSV-output (a friend taught me a while ago) like:
```
def generate_csv_lines(header, attribute_1, attribute_2, attribute_3, ...):
    """
    ...
    Returns
    -------
    csv_lines :
        A list whose elements are CSV "lines"
    """
    csv_lines = []
    csv_lines.append(str(header))
    csv_lines.append(attribute_1)
    ...
    return csv_lines
```

and then write out the "CSV" object.

**Comment by NikosAlexandris** [comment]
Also, note that it's not easy to automatically build a header for an CSV-output, from the dictionary (i.e. get selectively the "headers" from the 'metadata' dictionary). It would be nice to avoid handcrafting the header, somehow. Then again, it's easy to hardcode (i.e. `header = 'lat,lon,vv,vh'`) and I guess it wouldn't matter much (?).

**Comment by NikosAlexandris** [comment]
Fixed in 3d3419f081c4928dc00469b78239f0515e0dc6b7  -- Thank you so much @dimartinot
